frontend/src/entry/model/InferenceServices.tsx

```diff
@@ -15,6 +19,30 @@ type InferenceProps = {
 export default function InferenceServices({ model }: InferenceProps) {
   const router = useRouter()
   const { inferences, isInferencesLoading, isInferencesError } = useGetInferencesForModelId(model.id)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [healthCheck, setHealthCheck] = useState(false)
+
+  const { tokens, isTokensLoading, isTokensError, mutateTokens } = useGetUserTokens()
+  const { uiConfig, isUiConfigLoading, isUiConfigError } = useGetUiConfig()
+
+  useEffect(() => {
+    async function checkAuthentication() {
+      try {
+        setErrorMessage('')
+        const response = await fetch(`${uiConfig?.inference.connection.host}/api/health`, { credentials: 'include' })
+        setHealthCheck(response.ok)
+        if (!response.ok) {
+          setErrorMessage(await getErrorMessage(response))
+          return setErrorMessage('Login failed when when accessing the inferencing service')
```

Comment:
You're setting the error message twice here. The second will override the first. I'd suggest just doing
```
return setErrorMessage(await getErrorMessage(response))
```